tests/mocha/toolbox_test.js

```diff
@@ -5,6 +5,21 @@
  */
 
 suite('Toolbox', function() {
+  function convertToolBoxToJSONCaptureWarnings(toolboxXml) {
+    // TODO(#3985): Remove after warning log has been removed.
+    var toolboxJson;
+    var warnings = captureWarnings(() => {
+      toolboxJson = Blockly.utils.toolbox.convertToolboxToJSON(toolboxXml)
+    });
+    // convertToolboxToJSON doesn't always log a warning.
+    chai.assert.isTrue(warnings.length <= 1);
```

Comment:
@alschmiedt to comment on why this doesn't always log.

Reply:
It only throws a warning if the test is testing defining a toolbox using JSON.